unlock-app/src/utils/__mocks__/promises.js

```diff
@@ -0,0 +1,17 @@
+let x = 0
+
+const delayPromise = jest.fn(ms => {
+  if (x--) {
```

Comment:
This is not pure ;p
Not sure what `delayPromise` does actually?

Reply:
first of all, my main takeaway from your feedback is that the PR is not clear enough from a devel perspective, and this needs fixin!

To answer this specific point, the main issue that makes this PR so dang hard to do well is that the account is polled every half second.

The simplest solution is to use a `setInterval` but it turns out that this can wreak havoc on mobile browsers, according to the coinbase doc that Akeem and I used as a cheat sheet on our beginning exploration of this problem.

The next-most-simplest solution (setTimeout and infinitely looping) caused the tests to flail about like a fish and slowly croak when I tried to mock it with `jest.useFakeTimers()`. So, I moved to the delay promise, because it allows us to mock it out without using fake timers. A few commits ago, when it was passed as a parameter to the `walletService`, I could then directly pass in what this mock does, which is provide a delay promise that neither delays, nor loops.

In other words, I need a promise that will only successfully resolve once, and then reject. The rejection is the thing that kicks the account polling out of its infinite looping.

This mock module allows us also to specify the number of times that our delay will be called, in case we want to use it in the future and need more calls to succeed. Note that by default, the delay promise never resolves, and only rejects. This allows us to not generate an infinite loop in the other `walletService` tests, which prevents all kinds of weird side effects in those tests, mostly failing because the test doesn't mock a call to `web3.eth.getAccounts()`.

If this sounds incredibly complex, it's because it is. It took me 10 minutes to write the first working version of this, and has taken about 5 hours of work and 4+ days to get to this stage :). I'm not complaining, it's been a fun challenge, but it's a doozy, and I'm not surprised my code is not easy to read as a result.

Maybe we can chat briefly and look over the PR together for 10 minutes tomorrow, finalize how to solve/name/comment things and then I can implement any changes?